src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SlotCapacityService.kt

```diff
@@ -0,0 +1,71 @@
+package uk.gov.justice.digital.hmpps.visitscheduler.service
+
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.enums.VisitRestriction
+import uk.gov.justice.digital.hmpps.visitscheduler.exception.OverCapacityException
+import uk.gov.justice.digital.hmpps.visitscheduler.model.entity.session.SessionSlot
+import uk.gov.justice.digital.hmpps.visitscheduler.repository.SessionSlotRepository
+
+@Service
+@Transactional
+class SlotCapacityService {
+
+  companion object {
+    val LOG: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  @Autowired
+  private lateinit var sessionSlotRepository: SessionSlotRepository
+
+  @Autowired
+  private lateinit var visitService: VisitService
+
+  @Autowired
+  private lateinit var applicationService: ApplicationService
+
+  @Autowired
+  private lateinit var sessionTemplateService: SessionTemplateService
+
+  fun checkCapacityForBooking(sessionSlotReference: String, visitRestriction: VisitRestriction, incReservedApplications: Boolean) {
```

Comment:
should the visitRestriction be replaced with a SessionRestriction? also on the ApplicationDto and Entity. Maybe part of a different PR